hack/apply-helm-patches.sh

```diff
@@ -107,10 +107,28 @@ if [ -f "${PATCHES_DIR}/manager.yaml.patch" ]; then
                 cat "${PATCHES_DIR}/manager.yaml.patch" > "${TMP_PATCH}"
 
                 # Replace the whole args block with our patched version from the patch file
-                sed -i '/args:/,/command:/ {
-                    /command:/!d
-                    i\          args:\n            {{- range .Values.controllerManager.container.args }}\n            - {{ . }}\n            {{- end }}\n            - "--application-images-pull-policy={{ .Values.application.imagesPullPolicy }}"\n            - "--application-images-registry={{ .Values.application.imagesRegistry }}"\n            {{- if .Values.accessResources.traefik.enable }}\n            - "--watch-traefik"\n            {{- end}}
-                }' "${MANAGER_YAML}"
+                if [[ "$OSTYPE" == "darwin"* ]]; then
+                    # macOS sed requires empty string after -i
+                    sed -i '' '/args:/,/command:/ {
+                        /command:/!d
+                        i\
+          args:\
+            {{- range .Values.controllerManager.container.args }}\
+            - {{ . }}\
+            {{- end }}\
+            - "--application-images-pull-policy={{ .Values.application.imagesPullPolicy }}"\
+            - "--application-images-registry={{ .Values.application.imagesRegistry }}"\
+            {{- if .Values.accessResources.traefik.enable }}\
+            - "--watch-traefik"\
+            {{- end}}
```

Comment:
non-blocking/medium term: we should try to get rid of this altogether. See related comment on Henry's [PR](https://github.com/jupyter-infra/jupyter-k8s/pull/50#discussion_r2445053362)